src/macaron/repo_finder/commit_finder.py

```diff
@@ -0,0 +1,509 @@
+        parts.append(part)
+
+        numeric_only = numeric_only_pattern.match(part)
+
+        if not has_non_numeric_suffix and not numeric_only:
+            # A non-numeric part enables the flag for treating this and all remaining parts as version suffix parts.
+            # Within the built regex, such parts will be made optional.
+            # E.g.
+            # - 1.2.RELEASE -> 'RELEASE' becomes optional.
+            # - 3.1.test.2 -> 'test' and '2' become optional.
+            has_non_numeric_suffix = True
+
+        if count == len(split) - 1 and has_trailing_zero or has_non_numeric_suffix:
+            # This part will be made optional in the regex, hence the grouping bracket.
+            this_version_pattern = this_version_pattern + "("
+
+        if count == 1:
+            this_version_pattern = this_version_pattern + INFIX_1
+        elif count > 1:
+            this_version_pattern = this_version_pattern + INFIX_3
+
+        # Add the current part to the pattern.
+        this_version_pattern = this_version_pattern + part
+
+        if count == len(split) - 1 and has_trailing_zero or has_non_numeric_suffix:
+            # Complete the optional capture group.
+            this_version_pattern = this_version_pattern + ")?"
+
+    # If the version parts are less than MAX_ZERO_DIGIT_EXTENSION, add additional optional zeros to pad out the
+    # regex, and thereby provide an opportunity to map mismatches between version and tags (that are still the same
+    # number).
+    # E.g. MAX_ZERO_DIGIT_EXTENSION = 4 -> 1.2 to 1.2.0.0, or 3 to 3.0.0.0, etc.
+    if not has_non_numeric_suffix and 0 < len(parts) < MAX_ZERO_DIGIT_EXTENSION:
+        for count in range(len(parts), MAX_ZERO_DIGIT_EXTENSION):
+            # Additional zeros added for this purpose make use of a back reference to the first matched separator.
+            this_version_pattern = this_version_pattern + "(" + (INFIX_2 if count > 1 else INFIX_1) + "0)?"
+
+    this_version_pattern = (
+        f"^(?:(?:{PREFIX_WITH_SEPARATOR})|(?:{PREFIX_START}{name}{PREFIX_END}{PREFIX_SEPARATOR}))(?P<version>"
+        f"{this_version_pattern}){SUFFIX_SEPARATOR}{SUFFIX}$"
+    )
+    try:
+        return re.compile(this_version_pattern, flags=re.IGNORECASE), parts, has_non_numeric_suffix
+    except Exception as error:  # pylint: disable=broad-exception-caught
+        # The regex library uses an internal error that cannot be used here to satisfy pylint.
+        logger.debug("Error while compiling version regex: %s", error)
+        return None, [], False
+
+
+def match_tags(tag_list: list[str], artifact_name: str, artifact_version: str) -> list[str]:
+    """Return items of the passed tag list that match the passed artifact name and version.
+
+    Parameters
+    ----------
+    tag_list: list[str]
+        The list of tags to check.
+    artifact_name: str
+        The name of the artifact to match.
+    artifact_version: str
+        The version of the artifact to match.
```

Comment:
The only place I see this `match_tags` function being called is in the `get_commit_from_version` function, where the `name` and `version` params here are documented as "the name/version of the analysis target" rather than "of the artifact". I think it is better to be consistent when naming/documenting these parameters.